### PR TITLE
Handle the complex float in uniform and normal

### PR DESCRIPTION
**Pull request by JackCaoG**
verified `rand`, `uniform` and `normal` works. Thinking about adding formal c++ tests for complex types but that could be in a different pr. 
```
>>> x1 = torch.rand(2, 2, dtype=torch.complex64, device='xla:0')
>>> x1
tensor([[(0.5526+0.3920j), (0.3979+0.4389j)],
        [(0.2409+0.4744j), (0.3918+0.2904j)]], device='xla:0',
>>> x1.uniform_(0, 10)
tensor([[(8.9376+8.5000j), (9.1608+9.9034j)],
        [(7.9996+5.9927j), (5.2919+1.0915j)]], device='xla:0',
       dtype=torch.complex64)
>>> x1.normal_(1, 1)
tensor([[(0.7900+0.7625j), (1.9446+0.4353j)],
        [(1.3156+0.3884j), (0.7684-0.1768j)]], device='xla:0',
       dtype=torch.complex64)
```

### CODE REVIEW

File: torch_xla/csrc/random.cpp [hunk 1]
```diff
@@ -137,10 +149,12 @@ xla::XlaOp RngNormal(xla::XlaOp seed, const xla::Shape& shape, xla::XlaOp mean,
                                                GetBitGenerator(), rng_shape)
               .value;
       xla::XlaOp rng = xla::Complex(rng_real, rng_imag);
-      if (shape.element_type() == xla::PrimitiveType::C128) {
-        rng = xla::ConvertElementType(rng, xla::PrimitiveType::C128);
-      }
-      return XlaHelpers::PromotedAdd(mean, XlaHelpers::PromotedMul(rng, std));
+      // variance for normal distribution of the real and imaginary values is
+      // half of the input variance
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply torch_xla/csrc/random.cpp]: Yup.. should of take care of that when copying comments from pytroch..

File: torch_xla/csrc/random.cpp [hunk 2]
```diff
@@ -137,10 +149,12 @@ xla::XlaOp RngNormal(xla::XlaOp seed, const xla::Shape& shape, xla::XlaOp mean,
...
+      xla::XlaOp two = XlaHelpers::ScalarValue(2, XlaHelpers::TypeOfXlaOp(std),
+                                               rng_seed.builder());
+      return XlaHelpers::PromotedAdd(mean,
+                                     XlaHelpers::PromotedMul(rng, std / two));
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply torch_xla/csrc/random.cpp]: Yup... totally forgot about the sqrt part.

File: torch_xla/csrc/random.cpp [hunk 3]
```diff
@@ -35,60 +35,74 @@ xla::XlaOp MakeSeed(xla::XlaOp seed) {
   return xla::ConvertElementType(seed, xla::PrimitiveType::U64);
 }
 
+xla::XlaOp MakeUniformBoundaryValue(xla::XlaOp val) {
+  xla::PrimitiveType element_type = XlaHelpers::TypeOfXlaOp(val);
+  if (element_type == xla::PrimitiveType::BF16) {
+    return ConvertElementType(val, xla::PrimitiveType::F32);
```
Comment (dlibenzi) [inline torch_xla/csrc/random.cpp:41]: xla::

**Comment by dlibenzi** [comment]
Remember to squash ...

**Comment by JackCaoG** [comment]
> Remember to squash ...

Yup, will sqush and merge once the test finished running on TPU.

**Pull request merged**